src/Runner/Parallel/ProcessPool.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Runner\Parallel;
+
+use React\Socket\TcpServer;
+
+/**
+ * Represents collection of active processes that are being run in parallel.
+ * Inspired by {@see https://github.com/phpstan/phpstan-src/blob/ed68345a82992775112acc2c2bd639d1bd3a1a02/src/Parallel/ProcessPool.php}.
+ *
+ * @author Greg Korba <greg@codito.dev>
+ *
+ * @internal
+ */
+final class ProcessPool
+{
+    private TcpServer $server;
+
+    /** @var null|(callable(): void) */
+    private $onServerClose;
+
+    /** @var array<string, Process> */
+    private array $processes = [];
+
+    public function __construct(TcpServer $server, ?callable $onServerClose = null)
+    {
+        $this->server = $server;
+        $this->onServerClose = $onServerClose;
+    }
+
+    public function getProcess(ProcessIdentifier $identifier): Process
+    {
+        if (!\array_key_exists((string) $identifier, $this->processes)) {
+            throw ParallelisationException::forUnknownIdentifier($identifier);
+        }
+
+        return $this->processes[(string) $identifier];
+    }
+
+    public function addProcess(ProcessIdentifier $identifier, Process $process): void
+    {
+        $this->processes[(string) $identifier] = $process;
```

Comment:
The only consequence I can think of is worker not getting file chunks to process. Imagine that somehow there is more than one process with the same identifier - only last one will be recorded in the pool, but all of them would try to connect to the parallelisation operator and each time server would try to bind connection to known process (under specified identifier). I am not completely sure what would happen, but because `Runner` spawns all the processes at one (well, in a loop) and `ProcessIdentifier::create()` can't return the same identifier, I would not focus on it. This part also comes from PHPStan and they don't ensure uniqueness, so I'll leave it as-is and fix only if any problem occurs.